server/src/app.service.ts

```diff
@@ -143,5 +144,59 @@ export class AppService {
         round.roundId,
       );
     }
+
+  }
+  async enterResultsFromExternalDevice(enterResultsDto: EnterResultsDto) {
+    const round = await this.resultsService.getContestRound(
+      enterResultsDto.competitionWcaId,
+      enterResultsDto.eventId,
+      enterResultsDto.roundNumber,
+    );
+
+    for (const externalResultDto of enterResultsDto.results) {
+      const person = externalResultDto.wcaId
+        ? await this.personsService.getPersonByWcaId(externalResultDto.wcaId.toUpperCase())
+        : await this.personsService.getPersonById(externalResultDto.registrantId);
+
+      if (!person) throw new NotFoundException('Competitor not found');
+
+      const result: IResult | undefined = round.results.find(
+        (r) => r.personIds.length === 1 && r.personIds[0] === person.personId,
+      );
+      const roundFormat = roundFormats.find((rf) => rf.value === round.format);
+      const attempts: IAttempt[] = [];
+
+      for (let i = 0; i < roundFormat.attempts; i++) {
+        if (i < externalResultDto.attempts.length) {
+          attempts.push({ result: externalResultDto.attempts[i].result });
+        } else if (result?.attempts[i]) {
+          attempts.push(result.attempts[i]);
+        } else {
+          attempts.push({ result: 0 });
+        }
+      }
```

Comment:
I don't think you need any of this code here, since the attempts array is provided in whole, so you could just use attempts: enterResultsDto.attempts below, right? This code was necessary in the other function, because only one attempt was provided there.

Reply:
Removed

Reply:
It looks like you just removed the block with result: 0. I meant that you could remove this entire loop and use attempts: enterResultsDto.attempts below in the newResultPartial object.

Reply:
Oh okay, will do then